golang/internal/helper/docker/container_test.go

```diff
@@ -0,0 +1,185 @@
+		types.ContainerListOptions{All: true,
+			Filters: filters.NewArgs(
+				filters.KeyValuePair{
+					Key:   "name",
+					Value: fmt.Sprintf("^%s", testsuite.prefix),
+				})})
+	if err != nil {
+		log.Fatal().Err(err).Msg("")
+	}
+
+	timeoutValue := (time.Duration(dockerHelper.DockerClientTimeoutSeconds) * time.Second)
+	for i := range containers {
+		err = testsuite.dockerClient.ContainerStop(testsuite.ctx, containers[i].ID, &timeoutValue)
+		if err != nil {
+			log.Warn().Err(err).Msg("")
+		}
+		err = testsuite.dockerClient.ContainerRemove(testsuite.ctx, containers[i].ID, types.ContainerRemoveOptions{})
+		if err != nil {
+			log.Warn().Err(err).Msg("")
+		}
+	}
+}
+
+func (testsuite *DockerContainerHelperTestSuite) TestGetAllContainers() {
+	containers, err := dockerHelper.GetAllContainers(testsuite.ctx, nil)
+
+	assert.LessOrEqual(testsuite.T(), len(testsuite.testContainers), len(containers),
+		"should return at least as many containers as we started/created")
+	assert.Nil(testsuite.T(), err, "should not have errors")
+}
+
+func (testsuite *DockerContainerHelperTestSuite) TestGetAllContainersbyName() {
+	containers, err := dockerHelper.GetAllContainersByName(testsuite.ctx, nil, fmt.Sprintf("^%s", testsuite.prefix))
+
+	assert.Equal(testsuite.T(), len(testsuite.testContainers), len(containers),
+		"should return the containers we started/created")
+	assert.Nil(testsuite.T(), err, "should not have errors")
+}
+
+func (testsuite *DockerContainerHelperTestSuite) TestGetContainerbyNameFound() {
+	matched, err := dockerHelper.GetContainerByName(testsuite.ctx, nil, fmt.Sprintf("%s-nginx1", testsuite.prefix))
+
+	assert.Equal(testsuite.T(), fmt.Sprintf("/%s-nginx1", testsuite.prefix), matched.Names[0],
+		"should return the containerID of the created container")
+	assert.Nil(testsuite.T(), err, "should not have errors")
+}
+
+func (testsuite *DockerContainerHelperTestSuite) TestGetContainerbyNameNotFound() {
+	matched, err := dockerHelper.GetContainerByName(testsuite.ctx, nil, fmt.Sprintf("%s-nginx1", randstr.Hex(prefixLength)))
+
+	assert.Equal(testsuite.T(), "", matched.ID,
+		"should return empty string")
+	assert.Nil(testsuite.T(), err, "should not have errors")
+}
+
+func (testsuite *DockerContainerHelperTestSuite) TestStopAndRemoveContainer() {
+	contName := fmt.Sprintf("%s-%s", testsuite.prefix, testsuite.containerNames[0])
+	matched, err := dockerHelper.GetContainerByName(testsuite.ctx, nil, contName)
+
+	assert.Equal(testsuite.T(), "running", matched.State,
+		"should return a running container")
+	assert.Nil(testsuite.T(), err, "should not have errors")
+
+	err = dockerHelper.StopContainerByName(testsuite.ctx, nil, contName)
+
+	assert.Nil(testsuite.T(), err, "should not have errors")
+	matched, err = dockerHelper.GetContainerByName(testsuite.ctx, nil, contName)
+
+	assert.Equal(testsuite.T(), "exited", matched.State,
+		"should return a stopped container")
+	assert.Nil(testsuite.T(), err, "should not have errors")
```

Comment:
If you want to check the err is nil, use `assert.noError()` function.

Reply:
Good catch! Thanks! Replaced those.